### ISSUE TITLE
Unity + Firebase emulator suite: FirestoreSettings: Changing it at runtime crashes Unity

### ISSUE THREAD

**Issue by DavidZobrist** [firebase/quickstart-unity#719]
Hey :) We just set up the firebase emulator suite with success. 
But now we want to make our Unity project use that local instance. 

We tried
` FirebaseFirestore db = FirebaseFirestore.DefaultInstance;
db.Settings =  db.Settings.WithHost("localhost:8080").WithSslEnabled(false).WithPersistenceEnabled(false);`

But it crashes unity. Is there a way to set this yet in the Unity FIrebase SDK?

Also creating a new 
FirebaseFirestoreSettings objects  with the same values result in a crash when assigned to the current FirebaseFireStore.DefaultInstance.

**Comment by a-maurice** [comment]
Hi @DavidZobrist 

Unfortunately the emulator suite is currently not supported on Unity, though it is on our radar.  Sorry for the inconvenience.

**Comment by Glongooo** [comment]
Hello, @a-maurice,  have same needs, is there any news on emulator support?

**Comment by distastee** [comment]
+1 - Any plans or news here?

**Comment by raqso** [comment]
:+1: @a-maurice @chkuang-g It would be very useful.

**Comment by artoonie** [comment]
@a-maurice can you please note on the docs that Unity is not supported? It is not clear: https://firebase.google.com/docs/emulator-suite

**Comment by sevdanski** [comment]
@dconeybe I apologise, my previous message may have been a bit unclear.  My enquiry was more about the Auth, Functions and the Realtime Database emulators - I wasn't actually using Firestore at the time.

I have noticed that none of the unity tutorials appear to have set up instructions for any of these.  All of the node based tutorials seemed to make use of a UseEmulator method which does not appear to be in most of the unity apis.  (I did find that the FirestoreFunctions API has a method for setting it up with the emulator, and that is working)

The steps I have found online for the Database is to change the url in the google-services file, but that doesn't appear to be working for me.  Tracing out the database url in the settings always shows the default one.

I have found nothing on how to get it working with the Auth emulator at all - which I think I need to be able to have the auth properties appear in the functions emulator?

Aplogies if this was not the best place to ask these questions. I was repeatedly brought to this page for searches about how to get the emulators working in unity, and it seemed related, so I thought it was the best place to ask. If you could point me in the right direction regarding these issues, I would greatly appreciate it.

Also, for what it is worth, I was able to get Firestore working with the emulator using your above steps.

**Comment by sevdanski** [comment]
@DellaBitta @dconeybe  Thanks to you both for the info. I'll keep an eye out for the updates!

**Comment by mrchantey** [comment]
> though its on our roadmap.

Any chance we can get a peek at this roadmap? 
As developers we are making critical product decisions and knowing whether emulator support is around the corner or just a twinkle in googles eye would be incredibly helpful.

**Comment by StefanoCos** [comment]
Yeah it seems Unity firebase SDk is still not updated to fully support auth emulator, functions ( and maybe others?)

**Comment by pedro15** [comment]
We need a firebase update to support emulator throug Unity ASAP! is a must for testing

**Comment by cadcri** [comment]
We need it !!!

**Comment by ronald112** [comment]
it's been already 3 years since the first message was written, and still nothing..

**Comment by otri** [comment]
Holy cowza, how many people are blocked like we are, just because of a lack of resources at Firebase on Unity?  I would love some more transparency about what's working vs not working.  I hate wasting time hunting down what major pieces are not working.  Emulator suite is pretty darn important to get running inside a unit testing workflow.  Us Unity developers need it.

**Comment by pedro15** [comment]
PLEASE ADD IT

**Comment by Awais6** [comment]
Is there any Update?

**Comment by itsOliverBott** [comment]
So any news on the proper emulator support?
